activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -581,11 +581,23 @@ def load_additional_types(type_map, oids = nil) # :nodoc:
           end
         end
 
+        def prepared_statements?(sql)
```

Comment:
Do you think we could change this method name to something like `with_parameters?` since thats really what were checking for, and since I think the code 

```
if without_prepared_statement?(binds)
  if prepared_statements?(sql)
```

can read a little confusing.